### DISCUSSION TITLE
How to create a composite aspect, i.e. combining multiple aspects into a single one?

### DISCUSSION THREAD

**Discussion by ziriax** [metalama/Metalama#349] [Q&A] [answered]
Suppose I have many methods, all with the aspects `[Log]` and `[Cache]`.

How can I create a new aspect say called `QueryAttribute` that is the composite of these two?

**Comment by gfraiteur** [comment, marked as answer]
What we call _composite aspects_ are aspects that combine several pieces of advice. This is covered in the [conceptual documentation](https://doc.postsharp.net/metalama/conceptual/aspects/aspect-design). Basically you need to implement the `BuildAspect`.

However in your specific case, your implementation of `QueryAttribute.OverrideMethod` must combine the logic of both aspects in a single template.

Another approach is to have `QueryAttribute` provide `Log` and `Cache` as [child aspects](https://doc.postsharp.net/metalama/conceptual/aspects/child-aspects).

**Reply by ziriax** [reply]
Thanks for the prompt answer! Indeed I want to reuse the excellent aspect attributes you provide in the Patterns package, chaining several of these aspects into an aspect class. I will try the child approach.
